bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/handler/MeaterHandler.java

```diff
@@ -0,0 +1,167 @@
+            if (bridgeHandler != null) {
+                return bridgeHandler.getMeaterThings().get(config.getDeviceId());
+            }
+        }
+        return null;
+    }
+
+    private void update(Device meaterProbe) {
+        // Update all channels from the updated data
+        getThing().getChannels().stream().map(Channel::getUID).filter(channelUID -> isLinked(channelUID))
+                .forEach(channelUID -> {
+                    State state = getValue(channelUID.getId(), meaterProbe);
+                    updateState(channelUID, state);
+                });
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    private State getValue(String channelId, Device meaterProbe) {
+        Cook cook = meaterProbe.cook;
+        switch (channelId) {
+            case CHANNEL_INTERNAL_TEMPERATURE:
+                return new QuantityType<Temperature>(meaterProbe.temperature.internal, SIUnits.CELSIUS);
+            case CHANNEL_AMBIENT_TEMPERATURE:
+                return new QuantityType<Temperature>(meaterProbe.temperature.ambient, SIUnits.CELSIUS);
+            case CHANNEL_COOK_TARGET_TEMPERATURE:
+                if (cook != null) {
+                    return new QuantityType<Temperature>(cook.temperature.target, SIUnits.CELSIUS);
+                }
+                break;
+            case CHANNEL_COOK_PEAK_TEMPERATURE:
+                if (cook != null) {
+                    return new QuantityType<Temperature>(cook.temperature.peak, SIUnits.CELSIUS);
+                }
+                break;
+            case CHANNEL_COOK_ELAPSED_TIME:
+                if (cook != null) {
+                    return new QuantityType<>(cook.time.elapsed, Units.SECOND);
+                }
+                break;
+            case CHANNEL_COOK_REMAINING_TIME:
+                if (cook != null) {
+                    return new QuantityType<>(cook.time.remaining, Units.SECOND);
+                }
+                break;
+            case CHANNEL_COOK_ID:
+                if (cook != null) {
+                    return new StringType(cook.id);
+                }
+                break;
+            case CHANNEL_COOK_NAME:
+                if (cook != null) {
+                    return new StringType(cook.name);
+                }
+                break;
+            case CHANNEL_COOK_STATE:
+                if (cook != null) {
+                    return new StringType(cook.state);
+                }
+                break;
+            case CHANNEL_LAST_CONNECTION:
+                ZonedDateTime zdt = meaterProbe.getLastConnection();
+                if (zdt != null) {
+                    return new DateTimeType(zdt);
+                }
+            case CHANNEL_COOK_ESTIMATED_END_TIME:
+                if (cook != null) {
+                    if (cook.time.remaining > -1) {
+                        return new DateTimeType(ZonedDateTime.now().plusSeconds(cook.time.remaining));
```

Comment:
`TimeZoneProvider` should be used for correct time-zone configured in openHAB.

Reply:
Fixed!